ui/common/src/commonMain/kotlin/social/androiddev/common/utils/AsyncImage.kt

```diff
@@ -9,56 +9,29 @@
  */
 package social.androiddev.common.utils
 
-import androidx.compose.foundation.Image
-import androidx.compose.foundation.layout.Spacer
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.produceState
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.graphics.painter.Painter
 import androidx.compose.ui.layout.ContentScale
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
-import social.androiddev.common.network.util.runCatchingIgnoreCancelled
+import io.kamel.image.KamelImage
+import io.kamel.image.lazyPainterResource
 
 /**
  * Use this helper until we switch to a image loading library which supports multiplatform
  */
 @Composable
-fun <T> AsyncImage(
+fun AsyncImage(
     contentDescription: String,
     modifier: Modifier = Modifier,
     alignment: Alignment = Alignment.Center,
     contentScale: ContentScale = ContentScale.Fit,
-    load: suspend () -> T,
-    painterFor: @Composable (T) -> Painter,
+    url:String
 ) {
-    val image: T? by produceState<T?>(null) {
-        value = withContext(Dispatchers.IO) {
-            runCatchingIgnoreCancelled {
-                load()
-            }.fold(
-                onSuccess = { it },
-                onFailure = { t ->
-                    t.printStackTrace()
-                    null
-                }
-            )
-        }
-    }
-
-    if (image != null) {
-        Image(
-            painter = painterFor(image!!),
-            contentDescription = contentDescription,
-            contentScale = contentScale,
-            modifier = modifier,
-            alignment = alignment,
-        )
-    } else {
-        Spacer(
-            modifier = modifier
-        )
-    }
+    KamelImage(
```

Comment:
Is this truly KMP support or just JVM?

Reply:
just jvm, but ios has native support afaik

Reply:
Cool, we should follow-up on this and change this to an expect/actual